pkg/k8s/client.go

```diff
@@ -105,11 +108,50 @@ func CreateConfig() (*rest.Config, error) {
 	return createConfig(GetAPIServerURL(), GetKubeconfigPath(), GetQPS(), GetBurst())
 }
 
+func setDialer(config *rest.Config) func() {
+	context := (&net.Dialer{
+		Timeout:   30 * time.Second,
```

Comment:
Shouldn't this be `Timeout: option.Config.K8sHeartbeatTimeout`?

Reply:
This is just for dial timeouts, 30s was the kubelet hardcoded default. I don't think we want to set this the same as the heartbeat timeout as that would affect all client connections, especially in the case where you might set your timeout to be zero.

Reply:
Oh right!